.github/workflows/te-review.yml

```diff
@@ -0,0 +1,284 @@
+name: (TET) Build & Review & Acceptance tests
+on:
+  pull_request:
+    paths:
+      - 'backend/tet/**'
+      - 'frontend/tet/**'
+      - 'frontend/shared/**'
+      - 'frontend/*'
+      - '.github/workflows/te-review.yml'
+  workflow_dispatch:
+    inputs:
+      build_required:
+        description: "Build images (true/false)"
+        required: true
+        default: "false"
+      pr_number:
+        description: "Pull request number (if redeploy without build) or own number for environment"
+        required: true
+
+env:
+  CONTAINER_REGISTRY: ghcr.io
+  CONTAINER_REGISTRY_USER: ${{ secrets.GHCR_CONTAINER_REGISTRY_USER }}
+  CONTAINER_REGISTRY_PASSWORD: ${{ secrets.GHCR_TOKEN }}
+  CONTAINER_REGISTRY_REPO: ghcr.io/city-of-helsinki/${{ github.event.repository.name }}
+  REPO_NAME: ${{ github.event.repository.name }}
+  KUBECONFIG_RAW: ${{ secrets.KUBECONFIG_RAW }}
+  BUILD_ARTIFACT_FOLDER: 'build_artifacts'
+  SERVICE_ARTIFACT_FOLDER: 'service_artifacts'
+  BASE_DOMAIN: ${{ secrets.BASE_DOMAIN_STAGING }}
+  DATABASE_USER: user
+  DATABASE_PASSWORD: testing-password
+  K8S_REQUEST_CPU: 50m
+  K8S_REQUEST_RAM: 200Mi
+  K8S_LIMIT_CPU: 500m
+  K8S_LIMIT_RAM: 400Mi
+  K8S_PROBE_FAILURE_THRESHOLD: 30
+  K8S_PROBE_PERIOD: 20
+  EMPLOYER_URL: https://te-empl-${{ github.event.pull_request.number }}.${{ secrets.BASE_DOMAIN_STAGING }}
+  YOUTH_URL: https://te-yout-${{ github.event.pull_request.number }}.${{ secrets.BASE_DOMAIN_STAGING }}
+  ADMIN_URL: https://te-admn-${{ github.event.pull_request.number }}.${{ secrets.BASE_DOMAIN_STAGING }}
+  NEXT_PUBLIC_BACKEND_URL: https://te-bknd-${{ github.event.pull_request.number }}.${{ secrets.BASE_DOMAIN_STAGING }}
+  MOCK_FLAG: 1
+  NEXT_PUBLIC_SENTRY_DSN: ${{ secrets.NEXT_PUBLIC_SENTRY_DSN }}
+  NEXT_PUBLIC_SENTRY_ENVIRONMENT: ${{ secrets.NEXT_PUBLIC_SENTRY_ENVIRONMENT }}
+  CORS_ALLOW_ALL_ORIGINS: 1
+jobs:
+  build:
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        service: [ 'te-bknd', 'te-empl', 'te-yout', 'te-admn' ]
+        include:
+          - service: te-bknd
+            context: ./backend
+            dockerfile: ./backend/docker/tet.Dockerfile
+            port: 8000
+          - service: te-empl
+            context: ./frontend
+            dockerfile: ./frontend/tet/employer/Dockerfile
+            port: 3000
+          - service: te-yout
+            context: ./frontend
+            dockerfile: ./frontend/tet/youth/Dockerfile
+            port: 3000
```

Comment:
ports should be different. in docker-compose you have 3000, 3001, and 3002. You could use same here and other places too.

Reply:
Our reasoning was to use always port 3000 inside the frontend containers (least surprising) and then map to different ports outside the containers as needed. But we can of course change this so that all YJDH projects implement this in the same way.